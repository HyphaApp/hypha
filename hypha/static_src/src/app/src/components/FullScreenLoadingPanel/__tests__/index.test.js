import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import FullScreenLoadingPanel from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });
import LoadingPanel from '@components/LoadingPanel'


describe("Test full screen loading component", () => {
    
    const subject = mount(<FullScreenLoadingPanel
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and component", () => {
      expect(subject.find('.full-screen-loading-panel').length).toBe(1);
      expect(subject.containsMatchingElement(<LoadingPanel />)).toEqual(true);
    });
  
    test('render a full screen loading component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
