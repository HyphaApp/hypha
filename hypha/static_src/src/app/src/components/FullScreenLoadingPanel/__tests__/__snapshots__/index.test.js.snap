// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test full screen loading component render a full screen loading component 1`] = `
<FullScreenLoadingPanel>
  <div
    className="full-screen-loading-panel"
  >
    <LoadingPanel>
      <div
        className="loading-panel"
      >
        <div
          className="loading-panel__text"
        >
          <h5>
            Loading...
          </h5>
        </div>
        <div
          className="loading-panel__icon"
        >
          <OTFLoadingIcon>
            <svg
              className="logo-loading"
              viewBox="0 0 45 40"
            >
              <g
                fill="none"
                fillRule="nonzero"
              >
                <rect
                  className="logo-part logo-part-1"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(10 0)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-2"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(5 5)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-3"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(0 10)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-4"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(0 15)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-5"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(0 20)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-6"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(0 25)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-7"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(5 30)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-8"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(10 35)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-9"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(30 0)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-10"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(35 5)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-11"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(40 10)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-12"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(40 15)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-13"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(40 20)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-14"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(40 25)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-15"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(35 30)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-16"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(30 35)"
                  width="5"
                />
              </g>
            </svg>
          </OTFLoadingIcon>
        </div>
      </div>
    </LoadingPanel>
  </div>
</FullScreenLoadingPanel>
`;
